diff --git a/emberjson/_deserialize/reflection.mojo b/emberjson/_deserialize/reflection.mojo
index 3fa9c6a..22198b6 100644
--- a/emberjson/_deserialize/reflection.mojo
+++ b/emberjson/_deserialize/reflection.mojo
@@ -292,16 +292,16 @@ __extension SIMD(JsonDeserializable):
             else:
                 return Scalar[Self.dtype](Int(p.expect_bool()))
 
-        comptime if size > 1:
+        comptime if Self.size > 1:
             p.expect(`[`)
 
-        comptime for i in range(size):
+        comptime for i in range(Self.size):
             s[i] = parse_simd_element(p)
 
-            comptime if i < size - 1:
+            comptime if i < Self.size - 1:
                 p.expect(`,`)
 
-        comptime if size > 1:
+        comptime if Self.size > 1:
             p.expect(`]`)
 
     @staticmethod
diff --git a/emberjson/_serialize/reflection.mojo b/emberjson/_serialize/reflection.mojo
index 804c396..b305c9f 100644
--- a/emberjson/_serialize/reflection.mojo
+++ b/emberjson/_serialize/reflection.mojo
@@ -249,7 +249,7 @@ __extension String(JsonSerializable):
 
 __extension SIMD(JsonSerializable):
     def write_json(self, mut writer: Some[Serializer]):
-        comptime if size == 1:
+        comptime if Self.size == 1:
             comptime if Self.dtype.is_floating_point():
                 write_float(rebind[Scalar[Self.dtype]](self), writer)
             else:
@@ -257,8 +257,8 @@ __extension SIMD(JsonSerializable):
         else:
             writer.begin_array()
 
-            comptime for i in range(size):
-                comptime if i != size - 1:
+            comptime for i in range(Self.size):
+                comptime if i != Self.size - 1:
                     writer.write_item[True](self[i])
                 else:
                     writer.write_item[False](self[i])
